apps/playwright-browser-tunnel/src/PlaywrightBrowserTunnel.ts

```diff
@@ -0,0 +1,524 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { ChildProcess } from 'node:child_process';
+
+import type { BrowserServer, BrowserType, LaunchOptions } from 'playwright-core';
+import { WebSocket, type WebSocketServer } from 'ws';
+import semver from 'semver';
+
+import { TerminalProviderSeverity, TerminalStreamWritable, type ITerminal } from '@rushstack/terminal';
+import { Executable, FileSystem } from '@rushstack/node-core-library';
+
+/**
+ * Allowed Playwright browser names.
+ * @alpha
+ */
+export type BrowserNames = 'chromium' | 'firefox' | 'webkit';
+const validBrowserNames: Set<string> = new Set(['chromium', 'firefox', 'webkit']);
```

Comment:
```suggestion
const validBrowserNames: ReadonlySet<string> = new Set<BrowserName>(['chromium', 'firefox', 'webkit']);
```

I believe that widening should be allowed and enforce that the constituents of the set are in the expected list.

Reply:
Alternatively you can put `satisfies BrowserName[]` inside the constructor.